api/src/test/java/org/openmrs/api/impl/DiagnosisServiceTest.java

```diff
@@ -0,0 +1,202 @@
+
+package org.openmrs.api.impl;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.ConditionVerificationStatus;
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.Patient;
+import org.openmrs.api.DiagnosisService;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.BaseContextSensitiveTest;
+
+/**
+ * Test methods for {@link DiagnosisServiceImpl}
+ */
+public class DiagnosisServiceTest extends BaseContextSensitiveTest {
+
+	protected static final String DIAGNOSIS_XML = "org/openmrs/api/include/DiagnosisServiceImplTest-SetupDiagnosis.xml";
+
+	private DiagnosisService diagnosisService;
+
+	private EncounterService encounterService;
+
+	private PatientService patientService;
+
+	@Before
+	public void setup() {
+		if (diagnosisService == null) {
+			diagnosisService = Context.getDiagnosisService();
+		}
+		if (encounterService == null) {
+			encounterService = Context.getEncounterService();
+		}
+		if (patientService == null) {
+			patientService = Context.getPatientService();
+		}
+		executeDataSet(DIAGNOSIS_XML);
+	}
+
+	/**
+	 * Creates a diganosis object.
+	 *
+	 * @return diagnosis the diagnosis created
+	 */
+	private Diagnosis buildDiagnosis() {
```

Comment:
@pluwum 
What is the aim of this method? To instantiate a Diagnosis Object, right? 
I don't think it is necessary. That is what you do when you executeDataSet().